modules/apps/29-fee/keeper/keeper.go

```diff
@@ -212,29 +242,38 @@ func (k Keeper) GetAllPayees(ctx sdk.Context) []types.RegisteredPayee {
 
 // SetCounterpartyPayeeAddress maps the destination chain counterparty payee address to the source relayer address
 // The receiving chain must store the mapping from: address -> counterpartyPayeeAddress for the given channel
-func (k Keeper) SetCounterpartyPayeeAddress(ctx sdk.Context, address, counterpartyAddress, channelID string) {
-	store := ctx.KVStore(k.storeKey)
-	store.Set(types.KeyCounterpartyPayee(address, channelID), []byte(counterpartyAddress))
+func (k Keeper) SetCounterpartyPayeeAddress(ctx context.Context, address, counterpartyAddress, channelID string) {
+	store := k.storeService.OpenKVStore(ctx)
+	if err := store.Set(types.KeyCounterpartyPayee(address, channelID), []byte(counterpartyAddress)); err != nil {
+		panic(err)
+	}
 }
 
 // GetCounterpartyPayeeAddress gets the counterparty payee address given a destination relayer address
-func (k Keeper) GetCounterpartyPayeeAddress(ctx sdk.Context, address, channelID string) (string, bool) {
-	store := ctx.KVStore(k.storeKey)
+func (k Keeper) GetCounterpartyPayeeAddress(ctx context.Context, address, channelID string) (string, bool) {
+	store := k.storeService.OpenKVStore(ctx)
 	key := types.KeyCounterpartyPayee(address, channelID)
 
-	if !store.Has(key) {
+	has, err := store.Has(key)
+	if err != nil {
+		panic(err)
+	}
+	if !has {
 		return "", false
 	}
 
-	addr := string(store.Get(key))
-	return addr, true
+	addr, err := store.Get(key)
+	if err != nil {
+		panic(err)
+	}
+	return string(addr), true // TODO: why this cast?
```

Comment:
ditto on refactor to single `Get` call and check len bytes.

regarding the TODO - its stored plain string because its an address belonging to a counterparty chain and "in theory" could be non cosmos-sdk addr.

e.g. relayer submits MsgRecvPacket to chainB and looks up a stored address to encode into the acknowledgement. Ack is relayed back to source(chainA) and address is parsed from acknowledgement data and fee is paid out to the account on source chain.

Reply:
I believe the reasoning was likely around string usage being more human readable, rather than passing around the raw bytes of a foreign address. @AdityaSripal can probably speak to it more tbh